packages/venia-concept/src/CustomButton/CustomButton.js

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from 'react';
+import { oneOf, shape, string, bool } from 'prop-types';
+import { BrowserPersistence } from '@magento/peregrine/lib/util';
+
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+import defaultClasses from '@magento/venia-ui/lib/components/Button/button.css';
+
+const storage = new BrowserPersistence();
+
+const getRootClassName = (priority, negative) =>
+    `root_${priority}Priority${negative ? 'Negative' : ''}`;
+
+const CustomButton = props => {
+    const {
+        children,
+        classes: propClasses,
+        priority,
+        type,
+        negative,
+        disabled,
+        className,
+        ...restProps
+    } = props;
+
+    const storeCode = storage.getItem('store_view_code');
+    const [css, setCss] = useState(defaultClasses);
+    useEffect(() => {
+        if (storeCode === 'fr') {
+            loadCss();
+        }
```

Comment:
I'm not sure how I feel about components having an understanding of the store view code. I wonder if there'd be some mechanism of doing this at a framework level so we don't end up with store view codes in components. Just food for thought :)

Reply:
How would you know which css to load for which store code? Something will have to know, right? Also, this "CustomButton" is a custom component from the client, not built into PWA Studio. You can only do such extension through scaffolding so it's in venia-concept as an example.

Reply:
It really depends what we refer to when we say theming, this could mean changing one or two components, changing some CSS, or having a completely different tree of components with inheritance. 

Would be good to validate legitimate business cases of theme customization and see where they align.

Reply:
Yea, the point of this spike PR was just to explore dynamic CSS loading. My take is that customization and theming are going to continue to fall into the realm of our developer users. The extension framework allows for so much customization and granularity that it seems counter productive to implement some rigid structure around what _we_ consider theming.

Anyways, I agree that we need to understand the business cases before implementing anything concrete. I've written up my thoughts in the JIRA ticket.